File: 2015/Day1/day1.cpp
#include <iostream>
#include <fstream>

using namespace std;

int main()
{
    ifstream inData;
    char ch;
    int floor = 0, index = 0;

    inData.open("input.txt");

    while (inData)
    {
        inData >> ch;
        if (ch == '(')
            floor++;
        else if (ch == ')')
            floor--;

        ch = ' ';
        
        index++;

        if (floor == -1)
            cout << "Index: " << index << endl;
    }

    cout << floor;
}

File: 2015/Day1/day1.py
from operator import index


def main():
    floor = 0
    with open("2015/Day1/input.txt") as input:
        inData = input.read()

        for index, ch in enumerate(inData):
            if ch == '(':
                floor += 1
            elif ch == ')':
                floor -= 1

            if floor == -1:
                print(index)
        
        print(floor)




main()